Catrobat/Paintroid
# Paint-235: fix unit tests

* Fix false negative unit test in `BrushPickerIntegrationTest`
* Reenable radiobuttonsbehavior test as it's not failing anymore
* Refactor `BrushPickerIntegrationTest` to improve readability
* Remove redundant `SystemAnimations` calls in `ColorDialogIntegrationTest`
* Move `SystemAnimation` calls from `WelcomeActivityIntentsTestRule` to
  `IntroTestBase` class to remove duplication
* Refactor `ShapeToolIntegrationTest` to improve readability
* Fix `ShapeToolIntegrationTest` by adding `SystemAnimationsRule`
* ~~Let `SystemAnimations` throw an exception instead of logging an error
  Why: Logged errors are often ignored, the errors reface when unit
  tests fail afterwards. This is a permission problem, not a problem
  in the failed unit tests and should be visible early on.~~
* Move `testCreateManyLayers` to junit tests
  Why: This Test was created to prove a memory leak, the current
  implementation took way too long to test this, the new one
  does functionally the same directly in less time
* Remove `IntroTestBase` class and move functionality into rule / Subclasses
  Why: Test inheritance leads to unreadable and unmaintainable tests and
  should be avoided where possible
* Refactor `WelcomeActivityIntentsTestRule` to test if RTL settings have
  actually been set for the current context config and fail at an early
  point
  Why: RTL tests are failing silently on API 26 in our current approach,
  this should be visible early and should not lead to some random tests
  failing
~~* Refactor `updateLocale` to improve readability~~
~~* Remove `setLayoutDirection` call as it's already set in `setLocale`~~
* Remove old eclipse config files
* Fix ADB rule for jenkins
* Improve `DrawingSurface` locking
